### PR TITLE
Update DSTU 2 documentation for Medication Order.

### PR DESCRIPTION
**Pull request by jfrhong**
- Update supported search parameters
- Add terminology bindings
- Add note that contained Medications may be returned.

### CODE REVIEW

File: content/dstu2/medication-order.md [hunk 1]
```diff
@@ -7,33 +7,41 @@ title: MedicationOrder | DSTU 2 API
...
-`timing-boundsperiod-end`| N |[`date`](http://hl7.org/fhir/DSTU2/search.html#date)| The [stop time](http://hl7.org/fhir/DSTU2/MedicationOrder-definitions.html#MedicationOrder.dosageInstruction.timing) of the order. Must be prefixed by `le` (currently only support querying for orders completed prior to a certain time). Not accepted when querying active or draft orders. Example: `le2014-05-19T20:54:02.000Z`
-[`_count`](http://hl7.org/fhir/DSTU2/search.html#count)| N |[`number`](http://hl7.org/fhir/DSTU2/search.html#number)| The number of items to include in a page. Currently ignored if querying for active or draft statuses. Example: `50`
+ Name                 | Required? | Type        | Description
+----------------------|-----------|-------------|------------------------------------------------------------------
+`patient`             | Y         |[`reference`]| The identity of a patient to list orders for. Example: `12345`
+`status`              | Y         |[`token`]    | The status of the medication ordr, may be a list separated by commas. Example: `active,draft`
```
Comment (daliboz) [inline content/dstu2/medication-order.md]: "medication order" (spelling)

File: content/dstu2/medication-order.md [hunk 2]
```diff
@@ -7,33 +7,41 @@ title: MedicationOrder | DSTU 2 API
...
+`-timing-boundsPeriod`| N         |[`date`]     | The date-time which should fall within the [period] the medication should be given to the patient. Must be prefixed by 'ge'. Example: `ge2014-05-19T20:54:02.000Z`
```
Comment (daliboz) [inline content/dstu2/medication-order.md]: Does it really start with a dash?

**Comment by daliboz** [comment]
Other than comment, +1

File: lib/resources/dstu2/medication_order.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  description: How the medication is to be used by the patient.
+  children:
+
+  - name: additionalInstructions
+    required: 'No'
+    cardinality: 0..1
+    type: CodeableConcept
+    description: Supplemental instructions.
+    binding:
+        description: Identifies the medication being administered.
```
Comment (dennispatterson) [inline lib/resources/dstu2/medication_order.yaml]: This is the description we're using for medicationCodeableConcept

**Comment by dennispatterson** [comment]
👍 other than existing comments

File: content/dstu2/medication-order.md (hunk 1 above)
Comment (jfrhong) [inline content/dstu2/medication-order.md]: done.

File: content/dstu2/medication-order.md (hunk 2 above)
Comment (jfrhong) [inline content/dstu2/medication-order.md]: > Servers are able to declare additional parameters in the profiles referenced from their conformance statements. Servers should define search parameters starting with a "-" character to ensure that the names they choose do not clash with future parameters defined by this specification.

http://hl7.org/fhir/DSTU2/search.html#conformance

File: lib/resources/dstu2/medication_order.yaml (hunk 3 above)
Comment (jfrhong) [inline lib/resources/dstu2/medication_order.yaml]: Copy-pasta >_>

**Pull request merged**